ipatests/test_integration/test_sssd.py

```diff
@@ -233,3 +234,44 @@ def test_external_group_paging(self):
             # reset to original limit
             tasks.ldapmodify_dm(master, ldap_query.format(limit=orig_limit))
             sssd_conf_backup.restore()
+
+    def test_ext_grp_with_ldap(self):
+        """User and group with same name should not break reading AD user data.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4073
+
+        When aduser is added in extrnal group and this is group is added
+        in group with same name of nonprivate ipa user and possix id, then
+        lookup of aduser and group should be successful when cache is empty.
+        """
+        cmd = self.master.run_command(['sssd', '--version'])
+        sssd_version = platform_tasks.parse_ipa_version(
+            cmd.stdout_text.strip())
+        if sssd_version.version <= '2.2.2':
+            pytest.skip("Fix for sssd issue:4073 unavailable in 2.2.2")
+        client = self.clients[0]
+        user = 'ipatest'
+        id = '100996'
+        ext_group = 'ext-ipatest'
+        tasks.kinit_admin(self.master)
+        tasks.user_add(self.master, user, extra_args=[
+            '--noprivate', '--uid', id, '--gidnumber', id])
+        # add group with same name as user and id.
+        tasks.group_add(self.master, user, extra_args=['--gid', id])
+        tasks.group_add(self.master, ext_group, extra_args=['--external'])
+        self.master.run_command(
+            ['ipa', 'group-add-member', '--group', ext_group, user])
+        self.master.run_command([
+            'ipa', 'group-add-member', '--external',
+            self.users['ad']['name'], ext_group,
+            '--users=', '--groups='])
+        self.master.run_command(['getent', 'group', user])
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(client)
+        try:
+            result = client.run_command(['id', self.users['ad']['name']])
+            assert id in result.stdout_text
+            assert user in result.stdout_text
```

Comment:
It is better to use more specific check for the group name and id to be sure that false positive is not produced by presence of similar string elsewhere in the output.
You can use regexp like this one (needs checking):
```
exp_re = r'groups=[^=]*\b{id}\({group}\)'.format(id=id, group=group)
```